src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +23,147 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+s32 D_80AD6D10[] = { 0x48580064 };
+
+void BgLotus_Init(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    s32 pad;
+    s32 sp2C;
+
+    Actor_ProcessInitChain(&this->dyna.actor, (InitChainEntry*)D_80AD6D10);
+    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gLilyPadCol);
+    this->dyna.actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->dyna.actor.floorPoly, &sp2C,
+                                                               &this->dyna.actor, &this->dyna.actor.world.pos);
+    this->unk168 = 0x60;
+    this->dyna.actor.world.rot.y = ((s32)Rand_Next() >> 0x10);
+    this->actionFunc = func_80AD68DC;
+}
+
+void BgLotus_Destroy(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AD6830(Actor* thisx) {
+    BgLotus* this = THIS;
+
+    f32 temp_fv1;
+
+    if (thisx->params == 0) {
+        temp_fv1 = Math_SinF(this->unk166 * (M_PI / 4)) * ((0.014f * (this->unk166 / (80.0f * 0.1f))) + 0.01f);
+        thisx->scale.x = ((1.0f + temp_fv1) * 0.1f);
+        thisx->scale.z = ((1.0f - temp_fv1) * 0.1f);
+    }
+}
+
+void func_80AD68DC(BgLotus* this, PlayState* play) {
```

Comment:
If it is okay with you, I'd like to try removing the gotos (probably some combination of if else). Would you be willing to setup a scratch (I could also put general cleanup in the scratch as well)

Reply:
https://decomp.me/scratch/CQ9Sr

Reply:
Yeah, the function would look a bit better like this:
https://decomp.me/scratch/g6rM0

The goto's are actually just if-else statements and I added some cleanup in there too. hensldm can add anything I missed as well!

Reply:
Seems good. The only thing I would note is that `0.06544985f = M_PI / 48`